beetsplug/albumtypes.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+
+# This file is part of beets.
+# Copyright 2021, Edgars Supe.
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+
+"""Adds an album template field for formatted album types."""
+
+from __future__ import division, absolute_import, print_function
+
+from beets.autotag.mb import VARIOUS_ARTISTS_ID
+from beets.library import Album
+from beets.plugins import BeetsPlugin
+
+
+class AlbumTypesPlugin(BeetsPlugin):
+    """Adds an album template field for formatted album types."""
+
+    def __init__(self):
+        """Init AlbumTypesPlugin."""
+        super(AlbumTypesPlugin, self).__init__()
+        self.album_template_fields['atypes'] = self._atypes
+
+    def _atypes(self, item: Album):
+        self.config.add({
+            'types': [],
+            'ignore_va': [],
+            'brackets': '[]'
+        })
```

Comment:
This stuff should go in `__init__`; otherwise, it will try to set the defaults every time we use the `$atypes` field.

Reply:
Good point! 

Would it not be better to extract the config values (next lines after this) in `__init__()` as well, so it's not done every time `_atypes`is called? I see both approaches in other plugins, so I suppose it's not a big concern, but converting config to those structs isn't exactly free.

Reply:
That is less of a big deal, IMO, and it's fine to do it either way. The reason is that `config.add(...)` actually modifies `config`: it inserts a new "layer" with the default configuration values. So doing this every time actually adds multiple copies of that data. In contrast, `get` and `as_*` calls are read-only, so it doesn't matter as much where they appear.